src/python/py_generate_pipeline.cpp

```diff
@@ -25,8 +25,12 @@ using ov::genai::StringInputs;
 using ov::genai::TokenizedInputs;
 using ov::genai::Tokenizer;
 
+// When StreamerVariant is used utf-8 decoding is done by pybind and can lead to exception on incomplete texts.
+// Therefore strings decoding should be handled with PyUnicode_DecodeUTF8(..., "replace") to not throw errors.
+using PyBindStreamerVariant = std::variant<std::function<py::object(py::object)>, std::shared_ptr<StreamerBase>, std::monostate>;
```

Comment:
Why not
```suggestion
using PyBindStreamerVariant = std::variant<std::function<bool(std::string)>, std::shared_ptr<StreamerBase>, std::monostate>;
```
or stl's pybind analogs?

Reply:
With you suggestion `PyBindStreamerVariant` would be exactly equal to `StreamerVariant`. But if we leave `StreamerVariant` then while casing c++ std::string -> py::object / py::string for `lambda x: print(x)` it will raise exceptions for incomplete words.

Therefore object for python streamer should accept some python type as argument and we cast our std::string to python type manually.

>or stl's pybind analogs?
agreed with this one, changed to `std::variant<std::function<bool(py::str)>` to be more specific